bokeh/core/property/bases.py

```diff
@@ -371,7 +399,7 @@ def accepts(self, tp, converter):
         self.alternatives.append((tp, converter))
         return self
 
-    def asserts(self, fn, msg_or_fn):
+    def asserts(self, fn: Callable[[HasProps, T], bool], msg_or_fn: str | Callable[[HasProps, str, T], None]) -> Property[T]:
```

Comment:
Some type aliases would make this more readable

Reply:
I will be introducing type aliases when it will be clear what actually needs to be aliased, i.e. I need find common patterns of usage across bokeh, not just factor out common substrings. This will happen when sufficient type coverage is achieved. The process is iterative, so there will be multiple iterations before code like this will settle (similarly to how static typing was done in bokehjs).